gear-test/src/manager.rs

```diff
@@ -187,27 +203,39 @@ impl JournalHandler for InMemoryExtManager {
     }
     fn send_value(&mut self, from: ProgramId, to: Option<ProgramId>, value: u128) {
         if let Some(to) = to {
-            let mut actors = self.actors.borrow_mut();
-
-            if let Some(Some(actor)) = actors.get_mut(&from) {
+            if let Some(Some(actor)) = self.actors.get_mut(&from) {
                 if actor.balance < value {
                     panic!("Actor {:?} balance is less then sent value", from);
                 }
 
                 actor.balance -= value;
             };
 
-            if let Some(Some(actor)) = actors.get_mut(&to) {
+            if let Some(Some(actor)) = self.actors.get_mut(&to) {
                 actor.balance += value;
             };
         };
     }
 
-    fn store_new_programs(
-        &mut self,
-        _code_hash: gear_core::program::CodeHash,
-        _candidates: Vec<(ProgramId, MessageId)>,
-    ) {
-        todo!() // Patch 3 is coming
+    fn store_new_programs(&mut self, code_hash: CodeHash, candidates: Vec<(ProgramId, MessageId)>) {
+        if let Some(code) = self.codes.get(&code_hash).cloned() {
+            for (candidate_id, init_message_id) in candidates {
+                if !self.actors.contains_key(&candidate_id) {
+                    let program = Program::new(candidate_id, code.clone())
```

Comment:
Do we have two clones for `code` here, isn't it?

Reply:
Nope. Without `cloned` we can't mutate `self` in 226.